content/post/115-protocol-labs-and-igalia-collaborating-on-dweb-in-browsers.md

```diff
@@ -0,0 +1,194 @@
+
+## Browser extensions
+
+The same-origin limitation is obviously something desired for security reasons when web pages register custom handlers. However, when such a registration happens from browser extensions that are trusted by users, it makes sense to make an exception. For example, Firefox has allowed for a long time to declare custom handlers for `ipfs` and other protocols directly in the [WebExtension manifest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/protocol_handlers) and so without same-origin check.
+
+There were existing requests about this feature in Chromium bug trackers. To tackle this problem, the initial step has been to follow [Chromium project's process](https://chromium.googlesource.com/chromium/src/+/master/extensions/docs/new_api_proposal.md#Proposal-Process) and draft a [proposal](https://bugs.chromium.org/p/chromium/issues/detail?id=64100#c19) based on existing use cases and what Firefox implements.
+
+Since there is a strong preference within the Chromium project to rely on Web APIs for new extension features, code owners' counter-proposal has been to give more power to `registerProtocolHandler` in extension context ([cross-origin handler](https://chromium-review.googlesource.com/c/chromium/src/+/2287304) and [extension-specific schemes](https://chromium-review.googlesource.com/c/chromium/src/+/2560305)). The two patches for these landed recently in Chromium and will be available in version 89!
+
+For now, due to an existing [permission UI bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1079333), the protocol registration must happen in an extension tab, but you can already get an idea of the new possibility [on this video](https://vimeo.com/489748722) (make sure to enable subtitles for a detailed description):
+
+<div style="width: 640px; margin-left: auto; margin-right: auto">
+<iframe src="https://player.vimeo.com/video/489748722" width="640" height="361" frameborder="0" allow="autoplay; fullscreen" allowfullscreen></iframe>
+</div>
+
+## Secure Contexts
+
+The issue with redirected URLs not behaving like normal URLs is a bit more tricky and can come from several reasons. For local HTTP gateways such as the one provided by [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop#ipfs-desktop), one of the explanation is that browsers used not to treat these local URLs as insecure contexts and thus block various web platform features. This was [changed three years ago in Chrome](https://www.chromestatus.com/feature/6269417340010496) and specifications was updated accordingly. More precisely:
```

Comment:
s/and specifications was updated accordingly/and specifications were updated accordingly